scylla-cdc/src/consumer.rs

```diff
@@ -0,0 +1,474 @@
+        let mut end_of_batch = false;
+        let mut operation = OperationType::PreImage;
+        let mut ttl = None;
+
+        let data_count =
+            schema.mapping.len() + schema.deleted_mapping.len() + schema.deleted_el_mapping.len();
+        let mut data: Vec<Option<CqlValue>> = Vec::with_capacity(data_count);
+
+        for (i, column) in row.columns.into_iter().enumerate() {
+            if i == schema.stream_id {
+                stream_id = column.unwrap().into_blob().unwrap();
+            } else if i == schema.time {
+                time = column.unwrap().as_uuid().unwrap();
+            } else if i == schema.batch_seq_no {
+                batch_seq_no = column.unwrap().as_int().unwrap();
+            } else if i == schema.end_of_batch {
+                end_of_batch = column.unwrap().as_boolean().unwrap()
+            } else if i == schema.operation {
+                operation = OperationType::try_from(column.unwrap().as_tinyint().unwrap()).unwrap();
+            } else if i == schema.ttl {
+                ttl = column.map(|ttl| ttl.as_bigint().unwrap());
+            } else {
+                data.push(column);
+            }
+        }
+
+        CDCRow {
+            stream_id,
+            time,
+            batch_seq_no,
+            end_of_batch,
+            operation,
+            ttl,
+            data,
+            schema,
+        }
+    }
+
+    /// Allows to get a value from the column that corresponds to the logged table.
+    /// Returns None if there is no column with such name.
+    /// Returns Some(None) if such column exists, but its value is null.
+    /// Otherwise returns Some(&Some(x)) where x is the value in that column.
+    pub fn get_value(&self, name: &str) -> Option<&Option<CqlValue>> {
+        self.schema.mapping.get(name).map(|id| &self.data[*id])
+    }
+
+    /// Allows to get info if a value was deleted in this operation.
+    /// Returns None if there is no column with such name.
+    pub fn is_value_deleted(&self, name: &str) -> Option<bool> {
+        self.schema
+            .deleted_mapping
+            .get(name)
+            .map(|id| self.data[*id].is_some())
+    }
+
+    /// Allows to get deleted elements from a collection.
+    /// Returns None if there is no collection column with such name.
+    /// Returns Some(None) if there is such collection, but nothing was deleted from it.
+    /// Otherwise returns Some(Some(x)) where x is a reference to vector containing deleted values.
+    pub fn get_deleted_elements(&self, name: &str) -> Option<Option<&Vec<CqlValue>>> {
+        self.schema
+            .deleted_el_mapping
+            .get(name)
+            .map(|id| self.data[*id].as_ref().map(|val| val.as_set().unwrap()))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    // Because we are planning to extract a common setup to all tests,
+    // the setup for this module is based on generation fetcher's tests,
+    // which has already been merged to the main branch.
```

Comment:
The last part of this commit is really unnecessary (about merging stuff into `main`)